File: SpringLogicServer/src/main/java/dk/groupfive/SpringLogicServer/model/objects/ReviewFull.java
package dk.groupfive.SpringLogicServer.model.objects;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class ReviewFull implements IReview, Serializable {
    private ArrayList<ReviewItem> reviews;

    public ReviewFull() {
        /*reviews = new ArrayList<>();*/
    }

    @Override
    public double getRating() {
        long score = 0;
        for (ReviewItem item : reviews)
        {
            score += item.getRating();
        }
        return (double) score / reviews.size();
    }

    @Override
    public List<ReviewItem> getReviews() {
        return reviews;
    }

    public void addReview(ReviewItem reviewItem)
    {
        reviews.add(reviewItem);
    }

    @Override
    public String toString() {
        return "ReviewFull{" +
                "reviews=" + reviews +
                '}';
    }
}

File: DataServer/DataServer/Model/Obsolete/ReviewFull.cs
using DataServer.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace DataServer.Models
{
	public class ReviewFull : IReview
	{
		public List<ReviewItem> reviews { get; set; }

		public ReviewFull()
		{
			reviews = new List<ReviewItem>();
		}

		public double GetRating()
		{
			long score = 0;
            Console.WriteLine("Reviews.Count " + reviews.Count);
			foreach (ReviewItem item in reviews)
			{
				score += item.rating;
			}
            Console.WriteLine((double)(score / reviews.Count));
			return (double)score / reviews.Count;
		}

		public List<ReviewItem> GetReviews()
		{
			return reviews;
		}

		public void AddReview(ReviewItem review)
		{
			reviews.Add(review);
		}
	}
}
